From eae06103264e41122a402b752388fbfb4528e7f0 Mon Sep 17 00:00:00 2001
Message-Id: <eae06103264e41122a402b752388fbfb4528e7f0.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Fri, 14 Nov 2014 14:24:59 +0000
Subject: [PATCH 46/61] FOR_UPSTREAM [VPG]: drm/i915: Add total count to
 context status debugfs output

When there are lots and lots and even more lots of contexts (e.g. when running
with execlists) it is useful to be able to immediately see what the total
context count is.

Change-Id: If9726d4df86567100ecf53867b43f4753f08bf84
For: VIZ-1587
For: VIZ-4741
For: GMIN-3638
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 2ffab8d..0854a3f 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -2060,6 +2060,7 @@ static int i915_context_status(struct seq_file *m, void *unused)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct intel_engine_cs *ring;
 	struct intel_context *ctx;
+	uint32_t count = 0;
 	int ret, i;
 
 	ret = mutex_lock_interruptible(&dev->mode_config.mutex);
@@ -2070,12 +2071,14 @@ static int i915_context_status(struct seq_file *m, void *unused)
 		seq_puts(m, "power context ");
 		describe_obj(m, dev_priv->ips.pwrctx);
 		seq_putc(m, '\n');
+		count++;
 	}
 
 	if (dev_priv->ips.renderctx) {
 		seq_puts(m, "render context ");
 		describe_obj(m, dev_priv->ips.renderctx);
 		seq_putc(m, '\n');
+		count++;
 	}
 
 	list_for_each_entry(ctx, &dev_priv->context_list, link) {
@@ -2085,6 +2088,7 @@ static int i915_context_status(struct seq_file *m, void *unused)
 
 		seq_puts(m, "HW context ");
 		describe_ctx(m, ctx);
+		count++;
 		for_each_ring(ring, dev_priv, i) {
 			if (ring->default_context == ctx)
 				seq_printf(m, "(default context %s) ",
@@ -2113,6 +2117,8 @@ static int i915_context_status(struct seq_file *m, void *unused)
 		seq_putc(m, '\n');
 	}
 
+	seq_printf(m, "Total: %d contexts\n", count);
+
 	mutex_unlock(&dev->mode_config.mutex);
 
 	return 0;
-- 
1.7.9.5

